### DISCUSSION TITLE
配合paopaodns使用，paopao.dns解析不了

### DISCUSSION THREAD

**Discussion by hussion** [kkkgo/PaoPaoGateWay#53] [问和答]
> 1 从http://paopao.dns:7889/ppgw.ini下载配置。使用此方式，你需要配合你的DNS服务，把paopao.dns这个域名解析到你的静态文件> 服务IP，服务端口是7889。如果配合[PaoPaoDNS](https://github.com/kkkgo/PaoPaoDNS)使用，你只需要设置SERVER_IP参数和设> 置HTTP_FILE=yes，映射7889端口即可，你可以直接把配置文件ppgw.ini放在PaoPaoDNS的/data目录。

问题：ppdns网络使用的`macvlan`划分的独立IP，ppgw配合ppdns，按照文档进行配置，看ppgw的日志，paopao.dns解析不了，可能是什么原因？

**Comment by kkkgo** [comment]
paopao.dns的解析结果是SERVER_IP参数的值。你可以手动nslookup paopao.dns测试。

**Reply by kkkgo** [reply]
192.168.50.1是你ppdns的容器的IP吗？

**Reply by hussion** [reply]
不是，是路由器的IP

**Reply by kkkgo** [reply]
那肯定要下发ppdns的IP才能解析到。

**Reply by hussion** [reply]
具体该如何下发呢？路由器非openwrt，支持配置的功能有限，目前好像不支持自定义下发。另外请问下ppgw能否添加一个配置项，用户可指定加载ppgw.ini文件的URL ？这样应该就能比较好的解决这个问题了？

**Reply by kkkgo** [reply]
路由器不是可以设置DNS吗，普通路由器都有的功能吧。
docker定制network.ini本身也可以设置静态IP和DNS地址。
指定加载ppgw.ini文件的URL ： 就是定制ppgwurl.ini。参考docker定制镜像一节都有。

**Reply by kkkgo** [reply]
如果你不下发ppdns，那你的设备是怎么用上ppdns？
如果主路由连dns设置功能都没有的话，那静态路由功能估计也没了，那分流就搞不起来了。

**Reply by hussion** [reply]
我目前不想路由器设置全局dns哈，考虑到pt下载，以及一些其他诉求，目前是在需要科学的设备上手动设置dns来使用

**Reply by kkkgo** [reply]
DNS不影响pt下载

**Reply by hussion** [reply]
会影响，如果tracker域名被判定为走ppgw网关，就会被计算为"盒子"，会导致增加非正常流量，甚至有封号风险；而且有些tracker也不能设置为走国内解析 / 国外加密dns解析，都会因CF盾等原因导致打不开

**Reply by kkkgo** [reply]
加入trackerslist.txt就好了，跟CF盾没啥关系。

**Reply by hussion** [reply]
trackerlist并不保险，很难保证没有遗漏，而且维护也比较麻烦。直接物理隔离是最优方案

**Reply by kkkgo** [reply]
PT的tracker都是私有的，不存在遗漏的问题。BT的tracker大概率已经自动更新到里面去了，即使没在里面也不会走BT流量。

**Reply by hussion** [reply]
跟私有不私有没关系，当你手里有3,40个pt站，你就知道了，维护是个很大的问题，你没法保证tracker URL什么时候会更新和变化，一旦维护不及时，跑了流量账号都废了。而且直接隔离，完全就能避免这个问题。为什么还要冒风险和花精力去维护这个呢？

**Reply by kkkgo** [reply]
确实如果能直接隔离是很一劳永逸的方法。如果下PT是专门的机器/虚拟机，可以直接在机器上配置手动DNS，如果是docker就更简单，直接docker选项就可以设置某个容器的DNS。
